packages/request-client.js/test/in-memory-request.test.ts

```diff
@@ -0,0 +1,120 @@
+import { RequestNetwork } from '../src/index';
+import * as TestData from './data-test';
+
+import { http, HttpResponse } from 'msw';
+import { setupServer, SetupServer } from 'msw/node';
+import config from '../src/http-config-defaults';
+
+describe('handle in-memory request', () => {
+  let requestNetwork: RequestNetwork;
+  let spyPersistTransaction: jest.Mock;
+  let mockServer: SetupServer;
+
+  beforeAll(() => {
+    spyPersistTransaction = jest.fn();
+
+    mockServer = setupServer(
+      http.post('*/persistTransaction', ({ request }) => {
+        if (!request.headers.get(config.requestClientVersionHeader)) {
+          throw new Error('Missing version header');
+        }
+        return HttpResponse.json(spyPersistTransaction());
+      }),
+      http.get('*/getTransactionsByChannelId', () =>
+        HttpResponse.json({
+          result: { transactions: [TestData.timestampedTransactionWithoutPaymentInfo] },
+        }),
+      ),
+      http.post('*/ipfsAdd', () => HttpResponse.json({})),
+      http.get('*/getConfirmedTransaction', () => HttpResponse.json({ result: {} })),
+    );
+    mockServer.listen({ onUnhandledRequest: 'bypass' });
```

Comment:
Let's change this to `error`. If our tests are making calls to the mockServer that we haven't explicitly mocked, something is wrong.
```suggestion
    mockServer.listen({ onUnhandledRequest: 'error' });
```

Reply:
Nevermind. I rescind my comment. This is out of scope for this PR.